### PR TITLE
Search parameters

### PR DESCRIPTION
**Pull request by matboivin**
New formatting

Changed:
`filters=<Attribute>:<String>`
to:
`filters=<Attribute>:<Value>`

### CODE REVIEW

**Comment by Kerollmops** [comment]
Thank you @matboivin,

We are currently redesigning the filters syntax to make it support boolean operations (i.e. `AND`, `OR`, and `NOT`). When we have something functional we will write an entire documentation page to explain it and will probably ask you to review it :)

But yes, changing from `String` to `Value` is valid I suppose!

**Comment by matboivin** [comment]
@Kerollmops Thank you! I fix a few minor changes and I let this page in progress for now.

File: guides/advanced_guides/search_parameters.md [hunk 1]
```diff
@@ -151,33 +244,41 @@ $ curl -X GET -G 'http://localhost:7700/indexes/nzwlr302/search' \
...
-        -d filters='title:Kung%20Fu%20Panda' \
-        -i
+$ curl --request GET  -G 'http://localhost:7700/indexes/movies/search' \
+      -d q=shifu \
+      -d filters='title:Kung%20Fu%20Panda'
 ```
 
 ## Matches
 
+This setting takes a **Boolean value** (`true` or `false`) and defines whether the raw matches should be returned or not.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:257]:
```suggestion
This setting takes a **Boolean value** (`true` or `false`) and defines whether an object with information about the matches should be returned or not.
```

Comment (matboivin) [reply guides/advanced_guides/search_parameters.md:257]:
```suggestion
This setting takes a **Boolean value** (`true` or `false`) and defines whether an object that contain information about the matches should be returned or not.
```

Comment (bidoubiwa) [reply guides/advanced_guides/search_parameters.md]: Great!

File: guides/advanced_guides/search_parameters.md [hunk 2]
```diff
@@ -151,33 +244,41 @@ $ curl -X GET -G 'http://localhost:7700/indexes/nzwlr302/search' \
 }
 ```
 
-The parameter should be **URL-encoded**.
+The parameter shall be **URL-encoded**.
```
[the remark this answers is not in the document]
Comment (matboivin) [reply guides/advanced_guides/search_parameters.md]: Oops sorry I committed this change already.

File: guides/advanced_guides/search_parameters.md [hunk 3]
```diff
@@ -1,71 +1,146 @@
...
+| **[attributesToRetrieve](/guides/advanced_guides/search_parameters.md#attributes-to-retrieve)**   | Attributes to display in the returned documents                 |      `*`      |
+| **[attributesToCrop](/guides/advanced_guides/search_parameters.md#attributes-to-crop)**           | Attributes whose values have to be cropped                      |    `none`     |
+| **[cropLength](/guides/advanced_guides/search_parameters.md#crop-length)**                        | Length used to crop field values                                |     `200`     |
+| **[attributesToHighlight](/guides/advanced_guides/search_parameters.md#attributes-to-highlight)** | Attributes whose values will contain highlighted matching terms |    `none`     |
+| **[filters](/guides/advanced_guides/search_parameters.md#filters)**                               | Filter queries by an attribute value                            |    `none`     |
+| **[matches](/guides/advanced_guides/search_parameters.md#matches)**                               | Defines whether the raw matches should be returned or not       |    `false`    |
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md]:
```suggestion
| **[matches](/guides/advanced_guides/search_parameters.md#matches)**                               | Defines whether an object that contains information about the matches should be returned or not       |    `false`    |
```

File: guides/advanced_guides/search_parameters.md [hunk 4]
```diff
@@ -1,71 +1,146 @@
...
 
-`attributesToRetrieve=<Attribute>,<Attribute>,...`.
+Attributes to **display** in the returned documents.
+
+`attributesToRetrieve=<Attribute>,<Attribute>,...`
 
-Attributes that will appear in the returned documents.
+- `<Attribute>` (Optional, string, Defaults to `*`)
+
+  Comma-separated list of attributes that shall be in the returned documents.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:88]:
```suggestion
  Comma-separated list of attributes that will be in the returned documents.
```

Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:88]:
```suggestion
  Comma-separated list of attributes whose fields will be present in the returned documents.
```

File: guides/advanced_guides/search_parameters.md [hunk 5]
```diff
@@ -1,71 +1,146 @@
...
+
+#### Example
+
+If you want to get only the `overview` field values, set `attributesToRetrieve` to `overview`.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:92]:
```suggestion
If you want to get only the `overview` and `title` field and not the other fields, set `attributesToRetrieve` to `overview,title`.
```

File: guides/advanced_guides/search_parameters.md [hunk 6]
```diff
@@ -1,71 +1,146 @@
...
+
+```bash
+$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
+      -d q=shifu \
+      -d attributesToRetrieve=overview
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:97]:
```suggestion
      -d attributesToRetrieve=overview,title
```

File: guides/advanced_guides/search_parameters.md [hunk 7]
```diff
@@ -1,71 +1,146 @@
...
+```
 
 ## Attributes to crop
 
+Attributes whose values have to be cropped.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:102]:
```suggestion
Attributes whose values will be cropped if they contain a matched query word.
```

File: guides/advanced_guides/search_parameters.md [hunk 8]
```diff
@@ -1,71 +1,146 @@
...
+
 `attributesToCrop=<Attribute>,<Attribute>,...`
 
-Attributes of which the value will be cropped depending on the `cropLength` and the matches.
+- `<Attribute>` (Optional, string, defaults to empty)
+
+  Comma-separated list of attributes whose values shall be cropped according to the `cropLength` value.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:108]:
```suggestion
  Comma-separated list of attributes whose values will be cropped if they contain a matched query word. 
  
In the case a matched query word is found, the field's value will be cropped around the first matched query word according to the `cropLength` value (default `200` see below).
```

File: guides/advanced_guides/search_parameters.md [hunk 9]
```diff
@@ -1,71 +1,146 @@
...
 
 ::: tip
-This is useful when you have specific needs for displaying results on the front-end application.
+This is especially useful when you have to display content on the front-end in a specific way.
 :::
 
-**Cropping start at the first occurrence of the search query**. It only keeps `(cropLength - matchLength)/2` chars on each side of the first match.
+## Crop length
+
+Set a **limit on the length** of field values.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:116]: Could you add a link to the `see attributesToCrop` to the above title?
```suggestion
Set a length for the cropping around the matching query words (see attributesToCrop).
```

File: guides/advanced_guides/search_parameters.md [hunk 10]
```diff
@@ -1,71 +1,146 @@
...
+
+`cropLength=<Integer>`
+
+- `<Integer>` (Optional, positive integer, defaults to `200`)
+
+  The length used to crop field values.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:122]:
```suggestion
  The length used to crop field values around the first matched query word if the field's attribute is in the `attributedToCrop` list.
```

File: guides/advanced_guides/search_parameters.md [hunk 11]
```diff
@@ -1,71 +1,146 @@
...
+
+Cropping start at the **first occurrence of the search query**.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:124]:
```suggestion
Cropping start at the **first occurrence of the matched query word**.
```

File: guides/advanced_guides/search_parameters.md [hunk 12]
```diff
@@ -84,27 +159,29 @@ Our **cropped version is in the \_formatted object**.
 }
 ```
 
-## Crop length
-
-`cropLength=<Integer>`
-
-Total length of the cropped field. See [attributesToCrop](/guides/advanced_guides/search_parameters.md#attributes-to-crop)
+In the example above, the cropped version of `overview` is 200 characters longs. There are 100 characters before `shifu` (the first match) and 100 more characters from the first letter of `shifu`.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:162]:
```suggestion
In the example above, the cropped version of `overview` is 200 characters longs. There are 100 characters before `shifu` (the first match) and 100 more characters starting at the first letter of `shifu`.
```

File: guides/advanced_guides/search_parameters.md [hunk 13]
```diff
@@ -84,27 +159,29 @@ Our **cropped version is in the \_formatted object**.
...
 
 ## Attributes to highlight
 
+Attributes whose values will contain **highlighted matching terms**.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:166]:
```suggestion
Attributes whose values will contain **highlighted matching query words**.
```

File: guides/advanced_guides/search_parameters.md [hunk 14]
```diff
@@ -84,27 +159,29 @@ Our **cropped version is in the \_formatted object**.
...
+
 `attributesToHighlight=<Attribute>,<Attribute>,...`
 
-Every matching string sequence in the given attribute's field will be wrapped around an `<em>` tag
+- `<Attribute>` (Optional, string, defaults to empty)
+
+  Comma-separated list of attributes. Every matching string sequence in the given attribute field will be wrapped around an `<em>` tag.
```
Comment (bidoubiwa) [inline guides/advanced_guides/search_parameters.md:172]:
```suggestion
  Comma-separated list of attributes. Every matching query words in the given attribute field will be wrapped around an `<em>` tag.
```

File: guides/advanced_guides/search_parameters.md (hunk 3 above)
Comment (matboivin) [reply guides/advanced_guides/search_parameters.md]: Sorry, it was outdated as I replaced `shall`. I applied your changes.

File: guides/advanced_guides/search_parameters.md (hunk 4 above)
Comment (matboivin) [reply guides/advanced_guides/search_parameters.md]: Sorry, it was outdated as I replaced `shall`. I applied your changes.

**Pull request merged**